mcs/class/System.Data/Test/ProviderTests/System.Data.SqlClient/SqlConnectionTest.cs

```diff
@@ -287,12 +287,22 @@ public void Close ()
 		[Test]
 		public void ChangeDatabase ()
 		{
-			if (ConnectionManager.Instance.Sql.IsAzure)
-				Assert.Ignore("SQL Azure doesn't support 'ChangeDatabase'");
+			conn = new SqlConnection(connectionString);
+			conn.Open();
 
-			conn = new SqlConnection (connectionString);
-			conn.Open ();
-			conn.ChangeDatabase ("master");
+			bool isAzure = ConnectionManager.Instance.Sql.IsAzure;
+
+			try
+			{
+				conn.ChangeDatabase("master");
+			}
+			catch (SqlException exc) when (isAzure)
+			{
+				Assert.AreEqual(40508, exc.Number); //USE statement is not supported to switch between databases (Azure).
+				return;
+			}
+
+			Assert.IsFalse(isAzure);
```

Comment:
@EgorBo yep, that cleanup can be done in a separate PR :) The reason why the existing tests are written as `try { foo(); Assert.Fail(); } catch { }` is because we didn't have Assert.Throws in our old nunit fork.

Thankfully those dark days are behind us and we can use easier code now 😄